### PR TITLE
INFINITY-2707 Update docs to mention scale-in support

### PR DESCRIPTION
**Pull request by nickbp**
Dev guide:
- Describe how scale-in could be enabled by the developer

Ops guide:
- Move existing section for adding/removing pods adjacent to other pod operations
- Move pod operations section adjacent to plan operations section
- Update uninstall/reconfigure sections to reflect handling in 1.10+ EE
- Update DSE command examples to reflect current naming

### CODE REVIEW

File: docs/pages/developer-guide.md [hunk 1]
```diff
@@ -789,7 +789,7 @@ This generates the following Plan:
 }
 ```
 
-The step associated with instance 0 of the hello pod is never restarted and its step is initialized as COMPLETE.  Another step has been generated for instance 1. Once it has completed, the service will have transitioned from its previous configuration to the new target configuration.
+The step associated with instance 0 of the hello pod is never restarted and its step is initialized as COMPLETE.  Another step has been generated for instance 1. Once it has completed, the service will have transitioned from its previous configuration to the new target configuration. Note that by default, pods can be scaled up but not scaled down. Decreasing the number of pods count will result in a validation error when the Scheduler is restarted. As a safety measure, if you wish to allow scale-in of your pods, you must specify `allow-decommission: true` for each applicable pod.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "**Note:** By default, pods can be scaled up but not scaled down. Decreasing the number of pods will result in a validation error when the Scheduler is restarted. As a safety measure, if you wish to allow scale-in of your pods, you must specify `allow-decommission: true` for each applicable pod."

File: docs/pages/developer-guide.md [hunk 2]
```diff
@@ -1308,6 +1308,62 @@ $ py.test frameworks/helloworld/
 
 The most basic set of features present in the YAML representation of the `ServiceSpec` are [presented above](#service-spec). The remaining features are introduced below.
 
+### Pods
+
+You may specify the number of pod instances to be run for every pod. As a safety measure, after initial install users can increase but not decrease this value. If you wish to allow scale-in of your pods, you must specify `allow-decommission: true` for each applicable pod, like this:
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "after initial install, users can increase but not decrease this value."

remove ", like this"

File: docs/pages/developer-guide.md [hunk 3]
```diff
@@ -1308,6 +1308,62 @@ $ py.test frameworks/helloworld/
...
+```yaml
+name: "hello-world"
+pods:
+  hello:
+    count: 3
+    allow-decommission: true
+    ...
+```
+
+You should only enable this option if it is safe for the pod's tasks be destroyed, without needing to perform additional rebalancing or drain operations beforehand.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: remove comma after "destroyed"

File: docs/pages/developer-guide.md [hunk 4]
```diff
@@ -1308,6 +1308,62 @@ $ py.test frameworks/helloworld/
...
+
+Pods which are removed from a service specification entirely will be decommissioned.  All instances of undefined pods will have their tasks killed and all their resources released back to the cluster.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "Pods removed from a service...."

File: docs/pages/developer-guide.md [hunk 5]
```diff
@@ -1308,6 +1308,62 @@ $ py.test frameworks/helloworld/
...
+
+For example updating a `ServiceSpec` from:
```
Comment (sascala) [inline docs/pages/developer-guide.md]: Comma after "example"

Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 4 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 3 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 2 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 1 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

**Comment by nickbp** [comment]
ping, anyone wanna give a green checkmark?

**Pull request merged**